imagedephi/rules.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import TYPE_CHECKING
+
+import tifftools
+
+if TYPE_CHECKING:
+    from tifftools.tifftools import IFD, TagEntry
+
+
+@dataclass
+class TiffTagEntry:
+    tag: tifftools.TiffTag
+    entry: TagEntry
+
+
+class RuleType(Enum):
+    METADATA = "metadata"
+    IMAGE = "image"
+
+
+class RedactMethod(Enum):
+    REPLACE = "replace"
+    DELETE = "delete"
+    KEEP = "keep"
+
+
+class RuleFormat(Enum):
+    TIFF = "tiff"
+
+
+class RuleSource(Enum):
+    BASE = "base"
+    OVERRIDE = "override"
+
+
+@dataclass
+class Rule:
+    title: str
+    redact_method: RedactMethod
+    rule_type: RuleType
+
+    # Consider factory class fn here
+    # def make(...):
+
+    def get_title(self) -> str:
+        """Generate a title for the rule."""
+        return self.title
+
+
+@dataclass
+class TiffMetadataRule(Rule):
+    tag: tifftools.TiffTag
+    replace_value: str | bytes | list[int | float]
```

Comment:
I think we should allow `None` here, for anything other than `RedactMethod.REPLACE`. You could also further subclass this `RedactTiffMetadataRule`, but maybe that's too complicated.

Reply:
I implemented this in 50a1284. It led to a typing issue that I wasn't quite sure of the best solution to. We can't set the value of a tiff entry to `None`, but if a rule is using the `replace` method, and `replace_value` is `None`, that's what we'd be doing. 

We don't have validation on rules yet, so theoretically this could happen. My approach in the aforementioned commit is to just throw a runtime error, but maybe this should be walked back, and we just put in an empty string by default?

We could start doing some rule validation, and quit out early if we detect invalid override rules. Open to suggestions.